### PR TITLE
Added Settings

### PR DESCRIPTION
**Pull request by ketchupmann**

### CODE REVIEW

**Review by anthonymesa — CHANGES_REQUESTED** [review]
From now on, no more copying files from other pages to try to retrofit into something new. for every new page you should be starting with a new fresh empty file. Not only will this keep you from having random code in your files that doesn't fit/make sense, but it will also force you to actually learn how the pages are built in React.

Make sure that you go through the documentation to get examples and understanding of how react works: https://reactjs.org/docs/hello-world.html 

Before you continue to make any changes or fixes on this branch, always make sure you run
```
git pull origin main
```
to get the most recent changes from main that you may not have.

Make your changes, then push to this same branch. This pull request is linked to this branch, so when you make a new commit, leave a comment on the pull request letting me know you updated the branch.

File: src/settings/Settings.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import './Settings.css';
+import React from 'react';
+import logo from '../media/logo.svg';
+import { useNavigate } from "react-router-dom";
+
+//  Closest I can get to a javascript enum
+const FormType = {
+  User: Symbol("user"),
+  Bar: Symbol("bar")
+}
```
[the remark this answers is not in the document]
Comment (ketchupmann) [reply src/settings/Settings.js]: fixed

File: src/settings/Settings.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+const LoginHeader = () => {
+  return (
+    <div id="settings_header">
+        <div className="row header">
+        <p>Settings</p>
+        </div>
+    </div>
+  );
+}
```
[the remark this answers is not in the document]
Comment (ketchupmann) [reply src/settings/Settings.js]: fixed

File: src/settings/Settings.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+const invalidLoginAlert = () => {
+    alert("Current password is incorrect.");
+}
```
[the remark this answers is not in the document]
Comment (ketchupmann) [reply src/settings/Settings.js]: fixed

File: src/settings/Settings.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  let hashValue = await getHash(username, password);
+
+  let url = 'http://liquorish-server.azurewebsites.net/login/' + username + '/' + hashValue.toUpperCase();
+  console.log(url);
+  var xmlHttp = new XMLHttpRequest();
+  xmlHttp.open("GET", url, false); // false for synchronous request
+  xmlHttp.send(null);
+  
+  return xmlHttp.responseText;
+}
```
[the remark this answers is not in the document]
Comment (ketchupmann) [reply src/settings/Settings.js]: fixed

File: src/settings/Settings.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+          <button className="btnPrimary" onClick={ handleReset }>Reset</button>
+        </div>
+        <div className="row">
+            <button className="btnSecondary" onClick={ logoutUser }>Log Out</button>
+        </div>
+        <div className="row footer">
+           <div className="row">
+              <p>Saved Drinks</p>
+           </div>
+        </div>
```
[the remark this answers is not in the document]
Comment (ketchupmann) [reply src/settings/Settings.js]: fixed

File: src/settings/Settings.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    </div>
+  );
+}
+
+// end of additions for Settings A3
```
[the remark this answers is not in the document]
Comment (ketchupmann) [reply src/settings/Settings.js:166]: fixed

File: src/settings/Settings.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+  const completeLogin = async () => {
+    // need to get client ID and pass it up to index
+    navigate("home/user", { replace: true });
+  }
+
+  // validate the current password.
+  const handleSignIn = async () => {
+    const response = await validateLogin(username, inputcurrentpassword);
+    response === "true" ? completeLogin() : invalidLoginAlert();
+  }
```
[the remark this answers is not in the document]
Comment (ketchupmann) [reply src/settings/Settings.js]: fixed

File: src/settings/Settings.js [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+const LoginFormUser = (props) => {
+  let navigate = useNavigate();
+
+  // added for Settings A3
+  const [inputnewcityandstate, Setinputnewcityandstate] = React.useState("");
+  const [inputcurrentpassword, Setinputcurrentpassword] = React.useState("");
+  const [inputnewpassword, Setinputnewpassword] = React.useState("");
+  const [inputconfirmnewpassword, Setinputconfirmnewpassword] = React.useState("");
```
[the remark this answers is not in the document]
Comment (ketchupmann) [reply src/settings/Settings.js]: fixed

File: src/settings/Settings.js [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+        </div>
+        <div className="row">
+            <button className="btnSecondary" onClick={ logoutUser }>Log Out</button>
+        </div>
+        <div className="row footer">
+           <div className="row">
+              <p>Saved Drinks</p>
+           </div>
+        </div>
+    </div>
```
[the remark this answers is not in the document]
Comment (ketchupmann) [reply src/settings/Settings.js]: fixed

File: src/settings/Settings.js [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+           </div>
+        </div>
+        <div className="bodyContent">
+           <div className="row">
+              <p>City: Burlington, NC</p>
+           </div>
+            <div className="row">
+                <form>
+                    <input id="user_city_state" type="text" placeholder="(inputnewcityandstate)" ref={ usercitystate } onChange={ handleCityStateChange }/>
+                    <button className="btnPrimary" onClick={ handleCityState }>Search</button>
```
[the remark this answers is not in the document]
Comment (ketchupmann) [reply src/settings/Settings.js]: fixed

File: src/settings/Settings.css [hunk 11]
```diff
@@ -0,0 +1,373 @@
...
+}
+
+img[src="menu.svg"] {
+  margin: 20px;
+}
+
+.listItem {
+
+}
+/* #endregion */
```
[the remark this answers is not in the document]
Comment (ketchupmann) [reply src/settings/Settings.css]: fixed

File: src/settings/Settings.js [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+            <div className="row">
+                <input id="user_newpassword" type="password" placeholder="(inputnewpassword)" ref={ newPasswordInput } onChange={ handleNewPasswordChange }/>
+                <input id="user_confirmNewpassword" type="password" placeholder="(inputconfirmnewpassword)" ref={ confirmNewPasswordInput } onChange={ handleConfirmNewPasswordChange }/>
+            </div>
+        </div>
+        <div className="row">
+          <button className="btnPrimary" onClick={ handleReset }>Reset</button>
+        </div>
+        <div className="row">
+            <button className="btnSecondary" onClick={ logoutUser }>Log Out</button>
```
[the remark this answers is not in the document]
Comment (ketchupmann) [reply src/settings/Settings.js]: fixed

File: src/settings/Settings.js [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+            <div className="row">
+                <input id="user_currentpassword" type="password" placeholder="(inputcurrentpassword)" ref={ currentPasswordInput } onChange={ handleCurrentPasswordChange }/>
+            </div>
+            <div className="row">
+                <input id="user_newpassword" type="password" placeholder="(inputnewpassword)" ref={ newPasswordInput } onChange={ handleNewPasswordChange }/>
+                <input id="user_confirmNewpassword" type="password" placeholder="(inputconfirmnewpassword)" ref={ confirmNewPasswordInput } onChange={ handleConfirmNewPasswordChange }/>
+            </div>
+        </div>
+        <div className="row">
+          <button className="btnPrimary" onClick={ handleReset }>Reset</button>
```
[the remark this answers is not in the document]
Comment (ketchupmann) [reply src/settings/Settings.js]: fixed

**Review by anthonymesa — APPROVED** [review]

**Pull request merged**